app/controllers/admin/delivery_partners_controller.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+class Admin::DeliveryPartnersController < Admin::BaseController
+  skip_after_action :verify_authorized
+  skip_after_action :verify_policy_scoped
+
+  def choose_lead_providers
+    authorize DeliveryPartner, :create?
+
+    new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
+    if session[:delivery_partner_form].nil?
+      session[:delivery_partner_form] = { name: new_supplier_form.name }
+    else
+      session[:delivery_partner_form].merge!({ name: new_supplier_form.name })
+    end
+
+    @delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
+  end
+
+  def receive_lead_providers
+    authorize DeliveryPartner, :create?
+
+    provider_relationships = []
+    # Ensure all selected lead providers have at least one selected cohort
+    chosen_lead_providers = params.dig(:delivery_partner_form, :lead_providers)&.keep_if(&:present?)
+
+    unless chosen_lead_providers&.any?
+      @delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
```

Comment:
This seems to be done many times in many branches of `receive_lead_providers` - do we want to do it once before handling errors?